sdk/core/Azure.Core/tests/TestFramework/RecordedTestBase{TEnvironment}.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core.Testing
+{
+    public abstract class RecordedTestBase<TEnvironment> : RecordedTestBase where TEnvironment: TestEnvironment, new()
+    {
+        protected RecordedTestBase(bool isAsync) : base(isAsync)
+        {
+            TestEnvironment = new TEnvironment();
+        }
+
+        protected RecordedTestBase(bool isAsync, RecordedTestMode mode) : base(isAsync, mode)
+        {
+            TestEnvironment = new TEnvironment();
+        }
```

Comment:
For recorded test and samples should always use `.TestEnvironment`, `MyTestEnv.Instance` is currently used mostly by EventHubs and ServiceBus because they had an extensive test environment infrastructure before and to limit the size of this PR.